### PR TITLE
[bugfix] control override vaule by section

### PR DESCRIPTION
**Pull request by zhaoyongjie**

### CODE REVIEW

File: superset/assets/src/explore/components/ControlPanelsContainer.jsx [hunk 1]
```diff
@@ -89,17 +89,23 @@ class ControlPanelsContainer extends React.Component {
...
-          controls[control] &&
-          (
-            !controls[control].renderTrigger ||
-            controls[control].tabOverride === 'data'
-          )
-        )))) {
+        (control) => {
+          let controlByOverride = null;
+          if (controls[control] && controlOverrides[control]) {
+            controlByOverride = Object.assign({}, controls[control], controlOverrides[control]);
```
Comment (mistercrunch) [inline superset/assets/src/explore/components/ControlPanelsContainer.jsx]: We're moving away from `Object.assign`, please use the spread operator

File: superset/assets/src/explore/components/ControlPanelsContainer.jsx [hunk 2]
```diff
@@ -89,17 +89,23 @@ class ControlPanelsContainer extends React.Component {
...
     allSectionsToRender.forEach((section) => {
       if (section.controlSetRows.some(rows => rows.some(
-        control => (
-          controls[control] &&
-          (
-            !controls[control].renderTrigger ||
-            controls[control].tabOverride === 'data'
-          )
-        )))) {
+        (control) => {
```
Comment (mistercrunch) [inline superset/assets/src/explore/components/ControlPanelsContainer.jsx]: This code section is becoming even more opaque than it was before, and it was already bad before... For instance I'm confused between what's a control name and a control object, or object of objects, ... 

Also confused why we need to apply the overrides here where it should all be done in one place.

Comment (mistercrunch) [reply superset/assets/src/explore/components/ControlPanelsContainer.jsx]: I just checked and it appears the control overrides happen early on in `getControlsState`, why do we need it here too?

Comment (zhaoyongjie) [reply superset/assets/src/explore/components/ControlPanelsContainer.jsx]: hi, @mistercrunch, querySectionsToRender and displaySectionsToRender is dependent on renderTrigger/tabOverride properties in every control object,
before implement control override in VizType can not do overide here.
you can test this in Histogram diagram

**Pull request closed without merging**